src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,179 @@
+package duke;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Handles all interaction between the program and the files.
+ */
+public class Storage {
+
+    private String folderName;
+    private String fileName;
+    private String dirPath;
+    private String filePath;
+    private Ui ui = new Ui();
+
+    /**
+     * Initialise the path for the files.
+     * @param filePath The filePath that is needed to be stored.
+     */
+    public Storage(String filePath) {
+        String[] stringArr = filePath.split("/");
+        folderName = stringArr[0];
+        fileName = stringArr[1];
+        dirPath = System.getProperty("user.dir") + "\\src\\main\\" + folderName;
+        this.filePath = dirPath + "\\" + fileName;
+    }
+
+    /**
+     * Read the json file and initialise it into Duke program.
+     *
+     * @return The list of Task to be used in the program.
+     * @throws InitialisationError Failure to retrieve information from the files.
+     */
+    public List<Task> load() throws InitialisationError {
+        List<Task> taskList = new ArrayList<>();
+        try {
+            if (Files.notExists(Path.of(dirPath))) {
+                Files.createDirectories(Path.of(dirPath));
+            }
+            File txtDataFile = new File(filePath);
+            if (!txtDataFile.exists()) {
+                ui.printErrorMessage("No stored data! Starting a brand new state!");
+                txtDataFile = new File(filePath);
+                txtDataFile.createNewFile();
+            } else {
+                // file exist, read the file.
+                taskList = convertJsonToTaskListObj("dukeData.json");
+            }
+        } catch (Exception e) {
+            throw new InitialisationError("An error occurred while reading file: " + fileName);
+        }
+
+        return taskList;
+
+    }
+
+    /**
+     * Save and write data from the program into the files.
+     *
+     * @param taskList The list of task that will be saved.
+     */
+    @SuppressWarnings("unchecked")
+    public void saveData(TaskList taskList) {
+        File savedData = new File(filePath);
+
+        //Add employees to list
+        JSONArray jsonTaskList = new JSONArray();
+
+        for (int i = 0; i < taskList.getTotalNumberOfTask(); i++) {
+            Task currTask = taskList.getTaskById(i);
+            JSONObject taskDetails = new JSONObject();
+            String type = currTask.getType().toString();
```

Comment:
Should there be a newline between lines 90 and 91 to separate the code into separate logical chunks?